pootle/core/models/snapshot.py

```diff
@@ -90,8 +92,10 @@ def save(self, ctx_data):
         except OSError:
             pass
 
-        with open(self.filepath, 'w') as outfile:
-            outfile.write(ctx_data)
+        # use newline='' not to convert \n to OS-specific newlines;
+        # this is to be able to generate snapshots on any platform
+        with io.open(self.filepath, 'w', newline='') as outfile:
+            outfile.write(unicode(ctx_data))
```

Comment:
Thanks for the reference, that makes things clearer. The change it is referring to is in the built-in `open()` function, not in the `io.open()` method.

If it wasn't clear, I wasn't proposing to use `open()`, but `io.open()` **and** opening the file in binary mode, because what we have in the `ctx_data` is the result of [`json.dumps()`](https://docs.python.org/2/library/json.html#json.dumps), which is a byte string and there is no need to decode it into an `unicode` object, and then have it encoded again by the `write()` call. Simply write the bytes that we got, and done.